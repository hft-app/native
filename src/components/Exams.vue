<template>
    <div class="wrapper">
        <div v-if="exams.length === 0" class="screen container">
            <fa-icon class="icon-folder-open icon" icon="folder-open"/>
            <div class="title">Hier ist nichts</div>
            <div class="line"></div>
            <div class="info">
                <p>Bisher keine Prüfungsergebnisse.</p>
            </div>
        </div>
        <div v-else class="list">
            <div class="container">
                <div v-for="exam in exams" :class="examClass(exam) + ' exam article'">
                    <div v-if="!exam.grade && exam.passed" class="grade icon">
                        <fa-icon icon="check"/>
                    </div>
                    <div v-else-if="!exam.grade && !exam.passed" class="grade icon">
                        <fa-icon icon="times"/>
                    </div>
                    <div v-else class="grade icon">{{exam.grade}}</div>
                    <div class="data">
                        <div class="title">{{exam.title}}</div>
                        <div class="info">
                            <span v-if="exam.date">{{exam.date}} &middot;&nbsp;</span>
                            <span v-if="exam.cp">{{exam.cp}} CP &middot;&nbsp;</span>
                            <span v-if="exam.try === 1">Erstversuch</span>
                            <span v-else-if="exam.try === 2">Zweitversuch</span>
                            <span v-else>Drittversuch</span>
                        </div>
                    </div>
                </div>
            </div>
        </div>

        <div class="note">
            <div class="container">
                <p>Alle Angaben ohne Gewähr.</p>
            </div>
        </div>
    </div>
</template>
<script>
    export default {
        computed: {
            exams() {
                return this.$store.state.lsf.exams;
            }
        },
        methods: {
            examClass(exam) {
                return exam.passed ? 'passed' : 'failed'
            }
        }
    }
</script>
<style lang="scss" scoped>
    @import "../colors";

    .exam {
        display: flex;
        align-items: center;
        padding: 1em 0.2em;

        .grade {
            width: 2.4em;
            text-align: center;
            font-size: 2em;
            flex-shrink: 0;
        }

        &.passed .grade {
            color: #26A65B;
        }

        &.failed .grade {
            color: $primary;
        }

        .data {
            border-left: 1px solid $border;
            padding: 0 1em;
        }

        .title {
            margin-bottom: 6px;
        }

        .info {
            font-size: 14px;
            color: $secondary;
        }
    }
</style>
