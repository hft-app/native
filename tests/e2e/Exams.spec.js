import {config, createLocalVue, shallowMount} from '@vue/test-utils'
import Exams from 'components/Exams.vue';
import Vuex from 'vuex';

config.mocks['$t'] = str => str;

describe('Exams', () => {
  test('renders correctly', () => {
    const localVue = createLocalVue();
    localVue.use(Vuex);

    const store = new Vuex.Store({
      modules: {
        lsf: {
          state: {
            exams: [{
              passed: true,
              grade: 1.7,
              title: 'Exam0',
              date: new Date('Mon Feb 10 2020 08:00:00 GMT+0100'),
              cp: 3,
              try: 1
            }, {
              passed: false,
              grade: 5.0,
              title: 'Exam1',
              cp: 3,
              try: 2
            }, {
              passed: true,
              title: 'Exam2',
              cp: 3,
              try: 3
            }, {
              passed: false,
              title: 'Exam3',
              date: new Date('Mon Feb 10 2020 08:00:00 GMT+0100'),
              cp: 6,
              try: 2
            }]
          }
        }
      }
    });

    const wrapper = shallowMount(Exams, {
      stubs: ['fa-icon'],
      localVue,
      store
    });
    expect(wrapper.element).toMatchSnapshot()
  });
});



