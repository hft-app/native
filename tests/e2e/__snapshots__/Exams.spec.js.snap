// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exams renders correctly 1`] = `
<div
  class="wrapper"
>
  <div
    class="list"
  >
    <div
      class="container"
    >
      <div
        class="exam article passed"
      >
        <div
          class="grade icon"
        >
          1.7
        </div>
         
        <div
          class="data"
        >
          <div
            class="title"
          >
            Exam0
          </div>
           
          <div
            class="info"
          >
            <span>
              10.11.2020 · 
            </span>
             
            <span>
              3 CP · 
            </span>
             
            <span>
              page.exams.try[0]
            </span>
          </div>
        </div>
      </div>
      <div
        class="exam article failed"
      >
        <div
          class="grade icon"
        >
          5
        </div>
         
        <div
          class="data"
        >
          <div
            class="title"
          >
            Exam1
          </div>
           
          <div
            class="info"
          >
            <!---->
             
            <span>
              3 CP · 
            </span>
             
            <span>
              page.exams.try[1]
            </span>
          </div>
        </div>
      </div>
      <div
        class="exam article passed"
      >
        <div
          class="grade icon"
        >
          <fa-icon-stub
            icon="check"
          />
        </div>
         
        <div
          class="data"
        >
          <div
            class="title"
          >
            Exam2
          </div>
           
          <div
            class="info"
          >
            <!---->
             
            <span>
              3 CP · 
            </span>
             
            <span>
              page.exams.try[2]
            </span>
          </div>
        </div>
      </div>
      <div
        class="exam article failed"
      >
        <div
          class="grade icon"
        >
          <fa-icon-stub
            icon="times"
          />
        </div>
         
        <div
          class="data"
        >
          <div
            class="title"
          >
            Exam3
          </div>
           
          <div
            class="info"
          >
            <span>
              20.10.2020 · 
            </span>
             
            <span>
              6 CP · 
            </span>
             
            <span>
              page.exams.try[1]
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <div
    class="note"
  >
    <div
      class="container"
    >
      <p>
        page.common.noWarranty
      </p>
    </div>
  </div>
</div>
`;
